sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.State.cs

```diff
@@ -42,8 +42,7 @@ internal partial class Manager
         private void InitializeState()
         {
             SetPingState();
-            Task.Run(() => Run(CancellationToken.None)); // TODO: USE AN ACTUAL CANCELLATION TOKEN
-            // TODO: Investigate use of a dedicated thread here.
+            _thread.Start();
```

Comment:
Logic is scattered across different files making it very difficult to read the code.

Reply:
I moved all the state machine code into the `Manager.State` class